### PR TITLE
Channel fixes

### PR DESCRIPTION
**Pull request by Manishearth**
- Adds channel count modes
 - Adds channel interpretation modes
 - Adds all the upmix/downmix equations

Haven't yet written tests for this partially because I don't have quad/5.1 systems to test with.

r? @ferjm

### CODE REVIEW

File: servo-media/src/backends/gstreamer/audio_sink.rs [hunk 1]
```diff
@@ -124,7 +124,7 @@ impl AudioSink for GStreamerAudioSink {
             // sometimes nothing reaches the output
             if chunk.len() == 0 {
                 chunk.blocks.push(Default::default());
-                chunk.blocks[0].mix(channels as u8);
+                chunk.blocks[0].repeat(channels as u8);
```
Comment (sdroege) [inline servo-media/src/backends/gstreamer/audio_sink.rs]: You probably want to also signal the actual channel layout to GStreamer via the channel-mask or the positions array in the GstAudioInfo

File: servo-media/src/audio/block.rs [hunk 2]
```diff
@@ -126,15 +128,232 @@ impl Block {
...
+            // If we're discrete, have a repeat, and are downmixing,
+            // just truncate by changing the channel value
+            if self.repeat && self.channels > channels {
+                self.channels = channels;
+            } else {
+                // otherwise resize the buffer, silent-filling when necessary
+                self.resize_silence(channels);
+            }
+        } else {
+            // For speakers, we have to do special things based the
```
Comment (ferjm) [inline servo-media/src/audio/block.rs]: typo: "based on the"

File: servo-media/src/backends/gstreamer/audio_sink.rs (hunk 1 above)
Comment (Manishearth) [reply servo-media/src/backends/gstreamer/audio_sink.rs]: We only support one layout in gstreamer right now, yeah?

Comment (Manishearth) [reply servo-media/src/backends/gstreamer/audio_sink.rs]: Oh, I see. I'll be working on this when I add better support for output channel counts. Currently it's hardcoded to be two.

**Pull request merged**